public/app/percona/dbaas/components/DBCluster/DBClusterConnection/DBClusterConnection.tsx

```diff
@@ -20,12 +20,16 @@ export const DBClusterConnection: FC<DBClusterConnectionProps> = ({ dbCluster })
   const [loading, setLoading] = useState(true);
   const [connection, setConnection] = useState<ConnectionParams>(INITIAL_CONNECTION);
   const { result: clusters = {} } = useSelector(getPerconaDBClustersDetails);
-  const { result: dbClusters = [] } = useSelector(getPerconaDBClusters);
 
-  const { status, databaseType } =
-    Object.keys(clusters).length && dbCluster.id && dbClusters.length === Object.keys(clusters).length
+  const clusterInfo =
+    Object.keys(clusters).length && dbCluster.id
```

Comment:
I think you could avoid the additional check with clusterInfo if you modified the condition:

```
Object.keys(clusters).length && dbCluster.id && !!clusters[dbCluster.id]
```

This would return the status with unknown etc if the `clusters[dbCluster.id]` returned falsy value.

Reply:
fixed